### ISSUE TITLE
Fixed8 overflow in nep5 transfer method (neon-js v4.6.0)

### ISSUE THREAD

**Issue by dauTT** [CityOfZion/neon-js#450]
If we run this test using **neon-js v4.6.0**: 
```typescript

import Fixed8 from "../src/u/Fixed8";
import { transfer } from "../../neon-nep5/src/abi";

test("test3", () => {
  const account = new Account(
    "9ab7e154840daca3a2efadaf0df93cd3a5b51768c632f5433f86909d9b994a69"
  );

  const amtToSendTest = 922.33720369;
  const numOfDecimalsTest = 8;

  const generator = transfer(
    "9488220e8654d798f9b9cb9e74bd611ecc83fd0f",
    account.address,
    account.address,
    new Fixed8(amtToSendTest).div(Math.pow(10, 8 - numOfDecimalsTest))
  );
  const builder = generator();
});
```
we get the following errors: 
```typescript
Error: expected input to be less than 92233720368.54775807. Got input = 92233720369
```
The error happens in the `adjustedAmt` assignment, due to the fact that the `Fixed8` multiplication of `amt = 922.33720369` by 10^8 is greater than the maximum `Fixed8` value allowed which is 92233720368.54775807.

https://github.com/CityOfZion/neon-js/blob/master/packages/neon-nep5/src/abi.ts#L86

![image](https://user-images.githubusercontent.com/30392990/61404741-673d8c80-a8d8-11e9-9e15-4bd78ea80280.png)

This error was not present in **neon-js v4.5.2** because we were more relax there and we didn't have any boundary for the `Fixed8` value. The expected `Scriptbuilder` is supposed to be:

```
ScriptBuilder {
      str: '0531e28e79151435b20010db73bf86371075ddfba4e6596f1ff35d1435b20010db73bf86371075ddfba4e6596f1ff35d53c1087472616e73666572670ffd83cc1e61bd749ecbb9f998d754860e228894',
      pter: 0 }

(31e28e7915 = reverse hex of 92233720369)
```

I believe in this case `adjustedAmt` is actually the integer representation of `new u.Fixed8(amt)`, in other words

```typescript
adjustedAmt = new BN(new u.Fixed8(amt).toHex(), 16)
```
This will fix the `nep5 transfer method`. But perhaps we should create a method in the `Fixed8` class which provide us its integer representation. Do you agree @snowypowers ?

**Comment by snowypowers** [comment]
So what is happening is that the logic there is correct but we cannot do the multiplication within the Fixed8 model because we want to constraint the data model to what it means.

We still have to keep the multiplication so your fix should look like:
```
adjustedAmt = new BN(amt).mul(100000000);
```

To follow up, we could have a `toRawNumber` method that returns the actual underlying integer value as you suggested. 

This might not be the only place where we do this.

**Comment by dauTT** [comment]
Actually when we applies `toHex` the multiplication with 10^8 is implicitly done:
https://github.com/CityOfZion/neon-js/blob/master/packages/neon-core/src/u/Fixed8.ts#L76

so these two values are actually the same. 😄 
````typescript
new BN(new u.Fixed8(amt).toHex(), 16) === new BN(amt).mul(100000000)
````
Thanks for your feedback. I will provide a PR this evening.

**Comment by dauTT** [comment]
Closed via #451.

**Comment by klesun** [comment]
Hello. I don't know much about the context of this issue, but this was the only link in google with `expected input to be less than` error message, so maybe it blew up again somewhere few days ago (or was finally released? =-D).

Some of our users are reporting that they started getting this error whenever they try to perform a NEO transaction with either O3 or through in-house RPC request through [our dapp](https://blockchaincuties.com/login). Two of them were confirmed to be using Mac, maybe that has anything to do with the this.

![image](https://user-images.githubusercontent.com/5202330/94180886-5e1a5900-fea7-11ea-8137-d0c3e75a2558.png)

![image](https://user-images.githubusercontent.com/5202330/94180945-75594680-fea7-11ea-8376-f83532f1e287.png)

I don't request any investigation from your side, I can't reproduce the issue myself, just leaving this note here in case someone else also started having this issue recently.

P.S. this could also be something really stupid like us passing a decimal instead of integer units, will update if I find more info.
